datafusion/functions-aggregate/src/median.rs

```diff
@@ -230,6 +276,201 @@ impl<T: ArrowNumericType> Accumulator for MedianAccumulator<T> {
     }
 }
 
+/// The median groups accumulator accumulates the raw input values
+///
+/// For calculating the accurate medians of groups, we need to store all values
+/// of groups before final evaluation.
+/// So values in each group will be stored in a `Vec<T>`, and the total group values
+/// will be actually organized as a `Vec<Vec<T>>`.
```

Comment:
Given it is important to track the median values for each group separately I don't really see a way around Vec/Vec -- I think it is the simplest version and will have pretty reasonable performance

Reply:
Yes, I tried not to use `Vec<Vec<T>>` for avoiding copying from `Vec<Vec<T>>` to the `result Vec<T>`, but it is hard to do that.